### PR TITLE
Pullrequest improve contact status detection performance

### PR DESCRIPTION
**Pull request by BenSartor**

### CODE REVIEW

File: src/main/java/org/simlar/simlarserver/database/repositories/SubscriberRepository.java [hunk 1]
```diff
@@ -34,4 +35,7 @@ public interface SubscriberRepository extends CrudRepository<Subscriber, Long> {
 
     @Query("SELECT id FROM Subscriber WHERE username = ?1 AND domain = ?2")
     List<Long> findIdByUsernameAndDomain(final String username, final String domain);
+
+    @Query("SELECT username FROM Subscriber WHERE domain = ?1 AND username IN ?2")
```
Comment (jritzerfeld) [inline src/main/java/org/simlar/simlarserver/database/repositories/SubscriberRepository.java:39]: > Be careful when passing in many values. The JDBC standard does not guarantee that you can use more than 100 values for an IN expression list. Various databases exceed this number, but they usually have a hard limit for how many values are allowed. For example, Oracle’s limit is 1000.

https://docs.spring.io/spring-framework/reference/data-access/jdbc/parameter-handling.html#jdbc-in-clause

File: src/main/java/org/simlar/simlarserver/services/subscriberservice/SubscriberService.java [hunk 2]
```diff
@@ -124,11 +127,9 @@ public void checkCredentialsWithException(final String simlarId, final String ha
...
-    public int getStatus(final SimlarId simlarId) {
-        if (simlarId == null) {
-            return 0;
-        }
-
-        return subscriberRepository.findHa1ByUsernameAndDomain(simlarId.get(), sharedSettings.domain()).isEmpty() ? 0 : 1;
+    public List<SimlarId> filterSimlarIdsRegistered(final Collection<SimlarId> simlarIds) {
+        final Set<String> usernames = simlarIds.stream().map(SimlarId::get).collect(Collectors.toSet());
+        return subscriberRepository.findUsernameByDomainAndUsernameIn(sharedSettings.domain(), usernames)
+                .stream().map(SimlarId::create).toList();
```
Comment (jritzerfeld) [inline src/main/java/org/simlar/simlarserver/services/subscriberservice/SubscriberService.java:133]: Creating batches of ids should help limiting the number of arguments per database call. E.g. https://www.baeldung.com/java-stream-batch-processing#using-collection-api

Comment (BenSartor) [reply src/main/java/org/simlar/simlarserver/services/subscriberservice/SubscriberService.java]: Thank you again @jritzerfeld 
I have refined the implementation according to your suggestion.

**Review by jritzerfeld — APPROVED** [review]
It's Everything I Hoped For!

**Pull request merged**